pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -309,7 +309,11 @@ func (d *defaultDriver) newElasticsearchClient(
 	v version.Version,
 	caCerts []*x509.Certificate,
 ) esclient.Client {
-	url := services.ElasticsearchURL(d.ES, state.CurrentPodsByPhase[corev1.PodRunning])
+	url, err := services.ElasticsearchURLFromRandomPod(d.ES, state.CurrentPodsByPhase[corev1.PodRunning])
+	if err != nil {
+		log.V(1).Info("could not get a URL to communicate with elasticsearch from a random pod, failing back to using external service", "error", err)
+		url = services.ElasticsearchURL(d.ES, state.CurrentPodsByPhase[corev1.PodRunning])
```

Comment:
I'm not sure there is point to call `services.ElasticsearchURL` here as its only purpose is to call `ElasticsearchURLFromRandomPod` once again if `scheme` has changed, and we already know that `ElasticsearchURLFromRandomPod` returns an error.
```suggestion
		url = services.ExternalServiceURL(d.ES)
```